worlds/oot_soh/location_access/dungeons/forest_temple.py

```diff
@@ -2,8 +2,465 @@
+                                                                            can_do_trick(Tricks.FOREST_OUTDOORS_HEARTS_BOOMERANG, bundle)))
+    ])
+    # Connections
+    connect_regions(Regions.FOREST_TEMPLE_NW_OUTDOORS_LOWER, world, [
+        (Regions.FOREST_TEMPLE_LOBBY, lambda bundle: can_use(Items.SONG_OF_TIME, bundle)),
+        (Regions.FOREST_TEMPLE_COURTYARD_SKULLTULA_ISLAND, lambda bundle: can_use(Items.LONGSHOT, bundle)),
+        (Regions.FOREST_TEMPLE_NW_OUTDOORS_UPPER, lambda bundle: (can_do_trick(Tricks.HOVER_BOOST_SIMPLE, bundle) and
+                                                                  can_do_trick(Tricks.DAMAGE_BOOST_SIMPLE, bundle) and
+                                                                  has_explosives(bundle) and
+                                                                  can_use(Items.HOVER_BOOTS, bundle))),
+        (Regions.FOREST_TEMPLE_MAP_ROOM, lambda bundle: True),
+        (Regions.FOREST_TEMPLE_SEWER, lambda bundle: (has_item(Items.GOLDEN_SCALE, bundle) or
+                                                      can_use(Items.IRON_BOOTS, bundle))),
+        (Regions.FOREST_TEMPLE_BOSS_ENTRYWAY, lambda bundle: False)
+    ])
+    
+    ## Forest Temple NW Outdoors Upper
+    # Events
+    add_events(Regions.FOREST_TEMPLE_NW_OUTDOORS_UPPER, world, [
+        (EventLocations.FOREST_TEMPLE_DEKU_BABA_STICKS, Events.CAN_FARM_STICKS, lambda bundle: can_get_deku_baba_sticks(bundle)),
+        (EventLocations.FOREST_TEMPLE_DEKu_BABA_NUTS, Events.CAN_FARM_NUTS, lambda bundle: can_get_deku_baba_nuts(bundle)),
+    ])
+    # Locations
+    add_locations(Regions.FOREST_TEMPLE_NW_OUTDOORS_UPPER, world, [
+        (Locations.FOREST_TEMPLE_WEST_COURTYARD_RIGHT_HEART, lambda bundle: True),
+        (Locations.FOREST_TEMPLE_WEST_COURTYARD_LEFT_HEART, lambda bundle: True)
+    ])
+    # Connections
+    connect_regions(Regions.FOREST_TEMPLE_NW_OUTDOORS_UPPER, world, [
+        (Regions.FOREST_TEMPLE_NW_OUTDOORS_LOWER, lambda bundle: True),
+        (Regions.FOREST_TEMPLE_COURTYARD_SKULLTULA_ISLAND, lambda bundle: hookshot_or_boomerang(bundle)),
+        (Regions.FOREST_TEMPLE_BELOW_BOSS_KEY_CHEST, lambda bundle: True),
+        (Regions.FOREST_TEMPLE_FLOORMASTER_ROOM, lambda bundle: True),
+        (Regions.FOREST_TEMPLE_BLOCK_PUSH_ROOM, lambda bundle: True)
+    ])
+    
+    ## Forest Temple Courtyard Skulltula Island
+    # Locations
+    add_locations(Regions.FOREST_TEMPLE_COURTYARD_SKULLTULA_ISLAND, world, [
+        (Locations.FOREST_TEMPLE_GS_LEVEL_ISLAND_COURTYARD, lambda bundle: True)
+    ])
+    
+    ## Forest Temple NE Outdoors Lower
+    # Events
+    add_events(Regions.FOREST_TEMPLE_NE_OUTDOORS_LOWER, world, [
+        (EventLocations.FOREST_TEMPLE_STICKS, Events.CAN_FARM_STICKS, lambda bundle: can_get_deku_baba_sticks(bundle)),
+        (EventLocations.FOREST_TEMPLE_NUTS, Events.CAN_FARM_NUTS, lambda bundle: can_get_deku_baba_nuts(bundle))
```

Comment:
And the same here, call them `FOREST_TEMPLE_NE_OUTDOORS_LOWER_DEKU_BABA_STICKS` etc and add them to the EventLocations enum